packages/react-native-executorch/common/rnexecutorch/models/ocr/DetectorUtils.cpp

```diff
@@ -0,0 +1,654 @@
+    cv::threshold(textScoreComb, textScoreComb, 0.0, 1.0, cv::THRESH_TOZERO);
+    // Clamp values to be <= 1
+    cv::threshold(textScoreComb, textScoreComb, 1.0, 1.0, cv::THRESH_TRUNC);
+
+    // Dilate to connect word components
+    cv::Mat kernel = cv::getStructuringElement(cv::MORPH_RECT, cv::Size(3, 3));
+    cv::dilate(textScoreComb, textScoreComb, kernel, cv::Point(-1, -1), 2);
+  }
+
+  // 3. Find connected components to identify each character/word
+  cv::Mat binaryMat;
+  textScoreComb.convertTo(binaryMat, CV_8UC1);
+
+  cv::Mat labels, stats, centroids;
+  const int nLabels = cv::connectedComponentsWithStats(binaryMat, labels, stats,
+                                                       centroids, 4, CV_32S);
+
+  std::vector<DetectorBBox> detectedBoxes;
+  for (int i = 1; i < nLabels; ++i) { // Start from 1 to skip background
+    const int area = stats.at<int>(i, cv::CC_STAT_AREA);
+    if (area < 20) {
+      continue;
+    }
+
+    const int width = stats.at<int>(i, cv::CC_STAT_WIDTH);
+    const int height = stats.at<int>(i, cv::CC_STAT_HEIGHT);
+
+    // For vertical text (not single chars), height should be greater than width
+    if (!independentCharacters && height < width) {
+      continue;
+    }
+
+    // 4. For each component, create a mask and find its contour
+    cv::Mat mask = (labels == i);
+
+    // Create a segmentation map for the current component
+    cv::Mat segMap = cv::Mat::zeros(textMap.size(), CV_8U);
+    segMap.setTo(255, mask);
+
+    const int x = stats.at<int>(i, cv::CC_STAT_LEFT);
+    const int y = stats.at<int>(i, cv::CC_STAT_TOP);
+    const int w = stats.at<int>(i, cv::CC_STAT_WIDTH);
+    const int h = stats.at<int>(i, cv::CC_STAT_HEIGHT);
+
+    // Dynamically calculate dilation radius to expand the bounding box slightly
+    const int dilationRadius = static_cast<int>(
+        std::sqrt(static_cast<double>(area) / std::max(w, h)) * 2.0);
+    const int sx = std::max(x - dilationRadius, 0);
+    const int ex = std::min(x + w + dilationRadius + 1, imgW);
+    const int sy = std::max(y - dilationRadius, 0);
+    const int ey = std::min(y + h + dilationRadius + 1, imgH);
+
+    // Define a region of interest (ROI) and dilate it
+    cv::Rect roi(sx, sy, ex - sx, ey - sy);
+    cv::Mat kernel = cv::getStructuringElement(
+        cv::MORPH_RECT, cv::Size(1 + dilationRadius, 1 + dilationRadius));
+    cv::Mat roiSegMap = segMap(roi);
+    cv::dilate(roiSegMap, roiSegMap, kernel, cv::Point(-1, -1), 2);
+
+    // 5. Find the minimum area rotated rectangle around the contour
+    std::vector<std::vector<cv::Point>> contours;
+    cv::findContours(segMap, contours, cv::RETR_EXTERNAL,
+                     cv::CHAIN_APPROX_SIMPLE);
+
+    if (!contours.empty()) {
+      cv::RotatedRect minRect = cv::minAreaRect(contours[0]);
+
+      cv::Point2f vertices[4];
+      minRect.points(vertices);
+
+      std::array<Point, 4> points;
+#pragma unroll
+      for (int j = 0; j < 4; j++) {
```

Comment:
You can try eliminate one of magic numbers this way
```suggestion
      for (int j = 0; j < points.size(); j++) {
```